lib/lintPostcssResult.js

```diff
@@ -78,7 +78,7 @@ function lintPostcssResult(stylelintOptions, postcssResult, config) {
 			return;
 		}
 
-		const ruleSettings = get(config, ['rules', ruleName]);
+		const ruleSettings = config.rules && config.rules[ruleName];
 
 		if (ruleSettings === null || ruleSettings[0] === null) {
```

Comment:
[question] Thank you for addressing my suggestion! I have a question.
lodash's `get` returns `undefined` not `null`, doesn't it? I'm not sure why `ruleSettings === null` instead of `ruleSettings === undefined`... 🤔

Reply:
You're correct, lodash's `get` returns `undefined`. However, it seems that sometimes, `config.rules[ruleName]` can be set to `null`. I tried changing that line to check for `undefined` instead and got the test failures listed below.

I think we should leave that line alone.

<details>
<summary>Test failures without checking for null</summary>

```
 FAIL  lib/__tests__/standalone.test.js
  ✓ standalone with two file-specific globs (50 ms)
  ✓ standalone with input css (3 ms)
  ✓ standalone without input css and file(s) should throw error (13 ms)
  ✓ standalone with non-existent-file throws an error (7 ms)
  ✓ standalone with non-existent-file and allowEmptyInput enabled quietly exits (2 ms)
  ✓ standalone passing file with syntax error (1 ms)
  ✓ syntax error sets errored to true (1 ms)
  ✓ error `Cannot parse selector` sets errored to true (175 ms)
  ✓ configuration error sets errored to true (1 ms)
  ✓ unknown formatter option (1 ms)
  standalone with one input file
    ✓ triggers warning (166 ms)
  standalone with files and globbyOptions
    ✓ triggers warning (7 ms)
  standalone passing code with syntax error
    ✓ <input css 1> as source (2 ms)
    ✓ empty deprecations
    ✓ empty invalidOptionWarnings (1 ms)
    ✓ empty parseError (1 ms)
    ✓ error registered (1 ms)
    ✓ syntax error rule is CssSyntaxError (1 ms)
    ✓ syntax error severity is error
    ✓ (CssSyntaxError) in warning text (1 ms)
  standalone with different configs per file
    ✕ no warnings for A (42 ms)
    ✕ one warning for B (8 ms)
    ✕ correct warning for B (8 ms)
    ✕ no warnings for C (11 ms)
    ✕ no warnings for D (13 ms)
  standalone with config locatable from process.cwd not file
    ✓ two warning (25 ms)
    ✓ 'block-no-empty' correct warning (11 ms)
    ✓ 'plugin/warn-about-foo' correct warning (10 ms)
  configOverrides
    ✓ Setting `plugins` inside `**/__tests__/**` object should override the ones set in `config` object (7 ms)
    ✓ Setting `extends` inside `configOverrides` object should override the ones set in `config` object (2 ms)
  nonexistent codeFilename with loaded config
    ✓ does not cause error (1 ms)
    ✓ does load config from process.cwd (5 ms)
  existing codeFilename for nested config detection
    ✓ loads config from a nested directory (5 ms)

  ● standalone with different configs per file › no warnings for A

    TypeError: Cannot read property '0' of null

      81 | 		const ruleSettings = config.rules[ruleName];
      82 |
    > 83 | 		if (ruleSettings === undefined || ruleSettings[0] === undefined) {
         | 		                                              ^
      84 | 			return;
      85 | 		}
      86 |

      at forEach (lib/lintPostcssResult.js:83:49)
          at Array.forEach (<anonymous>)
      at lintPostcssResult (lib/lintPostcssResult.js:65:8)
      at lib/lintSource.js:96:13
          at async Promise.all (index 0)

  ● standalone with different configs per file › no warnings for A

    TypeError: Cannot read property 'find' of undefined

      244 |
      245 | 	it('no warnings for A', () => {
    > 246 | 		const resultA = results.find((result) => result.source.includes('a.css'));
          | 		                        ^
      247 |
      248 | 		expect(resultA.warnings).toHaveLength(0);
      249 | 	});

      at Object.<anonymous> (lib/__tests__/standalone.test.js:246:27)

  ● standalone with different configs per file › one warning for B

    TypeError: Cannot read property '0' of null

      81 | 		const ruleSettings = config.rules[ruleName];
      82 |
    > 83 | 		if (ruleSettings === undefined || ruleSettings[0] === undefined) {
         | 		                                              ^
      84 | 			return;
      85 | 		}
      86 |

      at forEach (lib/lintPostcssResult.js:83:49)
          at Array.forEach (<anonymous>)
      at lintPostcssResult (lib/lintPostcssResult.js:65:8)
      at lib/lintSource.js:96:13
          at async Promise.all (index 0)

  ● standalone with different configs per file › one warning for B

    TypeError: Cannot read property 'find' of undefined

      250 |
      251 | 	it('one warning for B', () => {
    > 252 | 		const resultB = results.find((result) => result.source.includes('b.css'));
          | 		                        ^
      253 |
      254 | 		expect(resultB.warnings).toHaveLength(1);
      255 | 	});

      at Object.<anonymous> (lib/__tests__/standalone.test.js:252:27)

  ● standalone with different configs per file › correct warning for B

    TypeError: Cannot read property '0' of null

      81 | 		const ruleSettings = config.rules[ruleName];
      82 |
    > 83 | 		if (ruleSettings === undefined || ruleSettings[0] === undefined) {
         | 		                                              ^
      84 | 			return;
      85 | 		}
      86 |

      at forEach (lib/lintPostcssResult.js:83:49)
          at Array.forEach (<anonymous>)
      at lintPostcssResult (lib/lintPostcssResult.js:65:8)
      at lib/lintSource.js:96:13
          at async Promise.all (index 0)

  ● standalone with different configs per file › correct warning for B

    TypeError: Cannot read property 'find' of undefined

      256 |
      257 | 	it('correct warning for B', () => {
    > 258 | 		const resultB = results.find((result) => result.source.includes('b.css'));
          | 		                        ^
      259 |
      260 | 		expect(resultB.warnings[0].text).toContain('Unexpected empty block');
      261 | 	});

      at Object.<anonymous> (lib/__tests__/standalone.test.js:258:27)

  ● standalone with different configs per file › no warnings for C

    TypeError: Cannot read property '0' of null

      81 | 		const ruleSettings = config.rules[ruleName];
      82 |
    > 83 | 		if (ruleSettings === undefined || ruleSettings[0] === undefined) {
         | 		                                              ^
      84 | 			return;
      85 | 		}
      86 |

      at forEach (lib/lintPostcssResult.js:83:49)
          at Array.forEach (<anonymous>)
      at lintPostcssResult (lib/lintPostcssResult.js:65:8)
      at lib/lintSource.js:96:13
          at async Promise.all (index 0)

  ● standalone with different configs per file › no warnings for C

    TypeError: Cannot read property 'find' of undefined

      262 |
      263 | 	it('no warnings for C', () => {
    > 264 | 		const resultC = results.find((result) => result.source.includes('c.css'));
          | 		                        ^
      265 |
      266 | 		expect(resultC.warnings).toHaveLength(0);
      267 | 	});

      at Object.<anonymous> (lib/__tests__/standalone.test.js:264:27)

  ● standalone with different configs per file › no warnings for D

    TypeError: Cannot read property '0' of null

      81 | 		const ruleSettings = config.rules[ruleName];
      82 |
    > 83 | 		if (ruleSettings === undefined || ruleSettings[0] === undefined) {
         | 		                                              ^
      84 | 			return;
      85 | 		}
      86 |

      at forEach (lib/lintPostcssResult.js:83:49)
          at Array.forEach (<anonymous>)
      at lintPostcssResult (lib/lintPostcssResult.js:65:8)
      at lib/lintSource.js:96:13
          at async Promise.all (index 0)

  ● standalone with different configs per file › no warnings for D

    TypeError: Cannot read property 'find' of undefined

      268 |
      269 | 	it('no warnings for D', () => {
    > 270 | 		const resultD = results.find((result) => result.source.includes('d.css'));
          | 		                        ^
      271 |
      272 | 		expect(resultD.warnings).toHaveLength(0);
      273 | 	});

      at Object.<anonymous> (lib/__tests__/standalone.test.js:270:27)
```

</summary>

Reply:
I did notice that `config.rules` will always be defined because this code is looping over its keys, so I've removed the extra check in b6314477.

Reply:
Never mind, TypeScript doesn't realize that `config.rules` will always be defined so I have to leave that check in. This is what I get for not running tests locally before I push. 😅